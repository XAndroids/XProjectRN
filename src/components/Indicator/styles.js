import {StyleSheet} from "react-native";

export const styles = StyleSheet.create({
    circleContainer: {
        flex: 1,
        flexDirection: 'row',
        justifyContent: 'center',
    },
    circle: {
        width: 10,
        height: 10,
        borderRadius: 5,
        backgroundColor: '#fff',
        marginHorizontal: 5
    },
    circleSelected: {
        backgroundColor: '#0f0'
    }
});
