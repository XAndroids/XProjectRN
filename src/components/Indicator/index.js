import React from 'react'
import {View} from 'react-native'
import PropTypes from 'prop-types'
import {styles} from "./styles"

const Indicator = (props) => {
    let circles = [];
    let {currentIndex, circleCount} = props;

    for (let i = 0; i < circleCount; i++) {
        let style = styles.circle;
        if (i === currentIndex) {
            style = [styles.circle, styles.circleSelected];
        }
        circles.push(<View testID={'indicator_view'} style={style} key={i}/>);
    }

    return (
        <View style={styles.circleContainer}>
            {circles}
        </View>
    )
};

Indicator.propTypes = {
    circleCount: PropTypes.number,
    currentIndex: PropTypes.number,
};

export default Indicator
