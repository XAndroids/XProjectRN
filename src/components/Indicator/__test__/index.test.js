import React from "react";
import Indicator from "../index";
import {render,shallow} from "react-native-testing-library";

describe("Indicator property test",()=>{
    it("circleCount property test",()=>{
        const circleCount = 3;
        const {getAllByTestId} = render(<Indicator circleCount={circleCount}/>)
        const indicatorViews = getAllByTestId('indicator_view')
        expect(indicatorViews.length).toBe(circleCount)
    })

    it("currentIndex property test",()=>{
        const circleCount = 3;
        const currentIndex = 1;
        const {getAllByTestId} = render(<Indicator circleCount={circleCount} currentIndex={currentIndex}/>)
        const indicatorViews = getAllByTestId('indicator_view')
        for(let view_i = 0; view_i <= indicatorViews.length; view_i ++){
            if(view_i === currentIndex){
                expect(indicatorViews[currentIndex].props.style[1].backgroundColor).toBe('#0f0')
            }else{
                expect(indicatorViews[currentIndex].props.style[0].backgroundColor).toBe('#fff')
            }
        }
    })
})

describe("Indicator Snapshot test",()=>{
    it("Indicator Snapshot test ",()=>{
        const circleCount = 3;
        const currentIndex = 1;
        const {toJSON} = render(<Indicator circleCount={circleCount} currentIndex={currentIndex}/>)
        expect(toJSON()).toMatchSnapshot()
    })
});

describe("Indicator Shallow test",()=>{
    it("Indicator Shallow test",()=>{
        const circleCount = 3;
        const currentIndex = 1;
        const {output} = shallow(<Indicator circleCount={circleCount} currentIndex={currentIndex}/>)
        expect(output).toMatchSnapshot()
    })
})
