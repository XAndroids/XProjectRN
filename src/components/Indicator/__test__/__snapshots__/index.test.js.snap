// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Indicator Shallow test Indicator Shallow test 1`] = `
<Component
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <Component
    style={
      Object {
        "backgroundColor": "#fff",
        "borderRadius": 5,
        "height": 10,
        "marginHorizontal": 5,
        "width": 10,
      }
    }
    testID="indicator_view"
  />
  <Component
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
          "borderRadius": 5,
          "height": 10,
          "marginHorizontal": 5,
          "width": 10,
        },
        Object {
          "backgroundColor": "#0f0",
        },
      ]
    }
    testID="indicator_view"
  />
  <Component
    style={
      Object {
        "backgroundColor": "#fff",
        "borderRadius": 5,
        "height": 10,
        "marginHorizontal": 5,
        "width": 10,
      }
    }
    testID="indicator_view"
  />
</Component>
`;

exports[`Indicator Snapshot test Indicator Snapshot test  1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "borderRadius": 5,
        "height": 10,
        "marginHorizontal": 5,
        "width": 10,
      }
    }
    testID="indicator_view"
  />
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
          "borderRadius": 5,
          "height": 10,
          "marginHorizontal": 5,
          "width": 10,
        },
        Object {
          "backgroundColor": "#0f0",
        },
      ]
    }
    testID="indicator_view"
  />
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "borderRadius": 5,
        "height": 10,
        "marginHorizontal": 5,
        "width": 10,
      }
    }
    testID="indicator_view"
  />
</View>
`;
